internal/provider/data_source_docker_image.go

```diff
@@ -0,0 +1,62 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDockerImage() *schema.Resource {
+	return &schema.Resource{
+		Description: "`docker_image` provides details about a specific Docker Image which need to be presend on the Docker Host",
+
+		ReadContext: dataSourceDockerImageRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of the Docker image, including any tags or SHA256 repo digests.",
+				Required:    true,
+			},
+			"latest": {
+				Type:        schema.TypeString,
+				Description: "The ID of the image in the form of `sha256:<hash>` image digest. Do not confuse it with the default `latest` tag.",
```

Comment:
In my understanding, the image digest is created when we push the image to the Docker Registry,
which means if the image isn't pushed to the Docker Registry, the image doesn't have image digest.
So I don't know whether it is good to deprecate the image id.

Reply:
The deprecation should be just renaming the confusing `latest` attribute.

If I've done a `docker build` on the local machine, and the image is created, it should still have a value for this.

Reply:
I've confirmed that although image id is stable from machine to machine, it's not actually the same as the image digest itself -- probably includes some extra metadata when the id is generated.

Reply:
> So I don't know whether it is good to deprecate the image id.

I'd suggest to remove the `latest` attribute and encourage folks to use `id` or `sha256_digest`

Reply:
yes, see 9ed8378f3106cbb2c8f7a5a7c974f3419a74f9fb

Reply:
Yes, I removed it in 9ed8378f3106cbb2c8f7a5a7c974f3419a74f9fb